valen-ui/src/main/java/be/twofold/valen/ui/viewers/Viewer.java

```diff
@@ -3,11 +3,13 @@
 import be.twofold.valen.core.game.*;
 import javafx.scene.*;
 
+import java.io.*;
+
 public interface Viewer {
 
-    boolean canPreview(AssetType type);
+    boolean canPreview(Asset asset);
 
-    void setData(Object data);
+    boolean setData(Asset asset, Archive archive) throws IOException;
```

Comment:
Why does it return a `boolean` as a way to handle errors but also throws an exception that is not thrown by half of the implementations of this interface?